File: fixtures/external-types/lib/tests/bindings/test_external_types.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import external_types_lib

// TODO: use an actual test runner.


// Test a round trip
do {
    let ct = getCombinedType(cval: CombinedType(
        cot: CrateOneType(sval: "test"),
        ctt: CrateTwoType(ival: 42)
    ))
    assert(ct.cot.sval == "test")
    assert(ct.ctt.ival == 42)
}

// Test passing in null value
do {
    let ct = getCombinedType(cval: nil)
    assert(ct.cot.sval == "hello")
    assert(ct.ctt.ival == 1)
}

File: fixtures/external-types/lib/tests/bindings/test_external_types.rb
# frozen_string_literal: true

# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/. */

require 'test/unit'
require 'external_types_lib'

class TestExternalTypes < Test::Unit::TestCase
  def test_round_trip
    ct = ExternalTypesLib.get_combined_type(ExternalTypesLib::CombinedType.new(
      ExternalTypesLib::CrateOneType.new("test"),
      ExternalTypesLib::CrateTwoType.new(42),
    ))
    assert_equal(ct.cot.sval, "test")
    assert_equal(ct.ctt.ival, 42)
  end

  def test_none_value
    ct = ExternalTypesLib.get_combined_type(nil)
    assert_equal(ct.cot.sval, "hello")
    assert_equal(ct.ctt.ival, 1)
  end
end
